### ISSUE TITLE
The config.kubernetes.io/local-config: "true" is too strict

### ISSUE THREAD

**Issue by kferrone** [kubernetes-sigs/kustomize#4124]
**Describe the bug**

When using the annotation `config.kubernetes.io/local-config: "true"`; you can not reference the resource. It is completely ignored. I would still like to have it as a referenced resource up until the final output when all with the annotation is removed. 

**Files that can reproduce the issue**

`pod.yaml`
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: buildup
  annotations:
    config.kubernetes.io/local-config: "true"
spec:
  containers:
    - name: app
      image: nginx
  restartPolicy: Always
```
`deployment.yaml`
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: buildup
```
`replacement.yaml`
```yaml
apiVersion: builtin
kind: ReplacementTransformer
metadata:
  name: buildup
replacements:
- source: 
    kind: Pod
    fieldPath: spec
  targets:
  - select:
      kind: Deployment
    fieldPaths: 
    - spec.template.spec
    options:
      create: true
```
`kustomization.yaml`
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
commonLabels:
  app.kubernetes.io/name: buildup
resources:
- pod.yaml
- deployment.yaml
transformers:
- replacement.yaml
```

**Expected output**

I expect only the deployment to be displayed with the values which were replaced from the Pod. The annotation is working in the sense it is excluding the resource in the final output, however it seems it was removed too soon as I can't reference it in the replacement. 

here is what the result should be:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.kubernetes.io/managed-by: kustomize-v4.2.0
    app.kubernetes.io/name: buildup
  name: buildup
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: buildup
  template:
    metadata:
      labels:
        app.kubernetes.io/name: buildup
    spec:
      containers:
      - image: nginx
        name: app
      restartPolicy: Always
```

**Actual output**

Basically the replacement transformer failed to find the Pod I told it to target. 
```
Error: nothing selected by ~G_~V_Pod|~X|~N:spec
```

**Kustomize version**

```
{Version:kustomize/v4.2.0 GitCommit:d53a2ad45d04b0264bcee9e19879437d851cb778 BuildDate:2021-07-01T00:44:28+01:00 GoOs:darwin GoArch:amd64}
```

**Platform**

all

**Additional context**

I can add the annotation after the replacement but then kustomize doesn't actually remove the resource from the output

**Comment by natasha41575** [comment]
I am in favor of keeping `local-config` annotated resources available to be used by transformers in the same kustomization layer, and see little reason why this would be undesirable. Unless other maintainers disagree, I am inclined to accept this as a feature request.
